SimpleWifi/AccessPoint.cs

```diff
@@ -138,24 +139,18 @@ public bool Connect(AuthRequest request, bool overwriteProfile = false)
 		/// <summary>
 		/// Connect asynchronous to the access point.
 		/// </summary>
-		public void ConnectAsync(AuthRequest request, bool overwriteProfile = false, Action<bool> onConnectComplete = null)
+		public async Task<bool> ConnectAsync(AuthRequest request, bool overwriteProfile = false, Action<bool> onConnectComplete = null)
```

Comment:
onConnectComplete Action is not used anymore. 
You can remove the parameter from the function signature.